plenum/test/view_change/test_new_node_joins_after_view_change.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+
+from plenum.test.pool_transactions.helper import \
+    disconnect_node_and_ensure_disconnected
+
+from plenum.test.node_catchup.helper import ensure_all_nodes_have_same_data
+from plenum.test.spy_helpers import getAllReturnVals
+from plenum.test.test_node import ensureElectionsDone, getNonPrimaryReplicas
+from plenum.test.view_change.helper import ensure_view_change, restart_node
+from stp_core.loop.eventually import eventually
+
+from plenum.test.helper import send_reqs_to_nodes_and_verify_all_replies, \
+    checkViewNoForNodes, stopNodes, sendReqsToNodesAndVerifySuffReplies
+from plenum.test.pool_transactions.conftest import clientAndWallet1, \
+    client1, wallet1, client1Connected, looper, stewardAndWallet1, steward1, \
+    stewardWallet
+from plenum.test.primary_selection.conftest import nodeThetaAdded, \
+    one_node_added
+
+
+@pytest.fixture(scope='module')
+def all_nodes_view_change(looper, txnPoolNodeSet, stewardWallet, steward1, client1, wallet1,
+          client1Connected):
+    for _ in range(5):
+        send_reqs_to_nodes_and_verify_all_replies(looper, wallet1, client1, 2)
+    ensure_view_change(looper, txnPoolNodeSet)
+    ensureElectionsDone(looper, txnPoolNodeSet)
+    ensure_all_nodes_have_same_data(looper, txnPoolNodeSet)
+
+
+@pytest.fixture(scope='module')
+def new_node_in_correct_view(all_nodes_view_change, looper, txnPoolNodeSet,
+                                      one_node_added, wallet1, client1):
+    new_node = one_node_added
+    looper.run(eventually(checkViewNoForNodes, txnPoolNodeSet, retryWait=1,
+                          timeout=10))
+    assert len(getAllReturnVals(new_node,
+                                new_node._start_view_change_if_possible,
+                                compare_val_to=True)) > 0
+    assert not new_node._next_view_indications
+    send_reqs_to_nodes_and_verify_all_replies(looper, wallet1, client1, 2)
+
+
+def test_new_node_has_same_view_as_others(new_node_in_correct_view):
```

Comment:
Because there is a fixture which completely describes the test, the test is just there to communicate what is being checked